### PR TITLE
Endpoint to obtain Loomies from an user

### PR DESCRIPTION
**Pull request by SilviaPabon**
To prove this, make a GET request to `/user/loomies`
Before, you must:
- Add a `owner` in some documents from collection `caught_loomies`
image
- Add the _id or ids in field `loomies` from `users` collection ( the id or ids are from the loomies which you add owner in the collection `caught_loomies`)

### CODE REVIEW

File: api/models/user.go [hunk 1]
```diff
@@ -208,3 +210,73 @@ func AddItemsToUserInventory(userId primitive.ObjectID, items []interfaces.GymRe
...
 	return nil
 }
+
+// GetLoomiesByUser returns an array of loomies according with user
+func GetLoomiesByUser(loomiesArray []primitive.ObjectID) ([]interfaces.UserLoomiesRes, error) {
+
+	// array to keep loomies ids
+	var loomiesIds []primitive.ObjectID = []primitive.ObjectID{}
+
+	for _, element := range loomiesArray {
```
Comment (PedroChaparro) [inline api/models/user.go:220]: I think this for loop is not necessary, `loomiesArray` is the same that `loomiesIds`.

File: api/models/user.go [hunk 2]
```diff
@@ -208,3 +210,73 @@ func AddItemsToUserInventory(userId primitive.ObjectID, items []interfaces.GymRe
 
 	return nil
 }
+
+// GetLoomiesByUser returns an array of loomies according with user
+func GetLoomiesByUser(loomiesArray []primitive.ObjectID) ([]interfaces.UserLoomiesRes, error) {
```
Comment (PedroChaparro) [inline api/models/user.go:215]: Maybe you can change the name to `GetLoomiesByIds`, the user isn't an argument of the function.

**Review by PedroChaparro — CHANGES_REQUESTED** [review]

**Review by PedroChaparro — APPROVED** [review]

File: api/controllers/user.go [hunk 3]
```diff
@@ -185,6 +187,38 @@ func HandleNewCodeValidation(c *gin.Context) {
...
+
+	user, err := models.GetUserById(userid.(string))
+
+	// user exists or not
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"message": "User was not found"})
+			return
+		} else {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"message": "Internal server error"})
```
Comment (Woynert) [inline api/controllers/user.go:202]: I believe the format we're going for is:

```
{
	"error": boolean,
	"message": string
}
```

I think @PedroChaparro can shed some light into this.

Comment (PedroChaparro) [reply api/controllers/user.go:202]: Yes, that's correct. The idea is to make all the endpoints to respond with the same format, at this way the client can trust that if there is some error (`"error": true`), there should be also a message to show using an alert or something like that (`"message": "The message"`).

Comment (SilviaPabon) [reply api/controllers/user.go:202]: Sure, this PR was also open, I will change this in it as well! Thanks!

**Review by Woynert — APPROVED** [review]

**Pull request merged**